lib/dao.js

```diff
@@ -2632,6 +2632,8 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
           'replaceById() must be implemented by the connector');
 
   var pkName = idName(this);
+  var initialData = {};
+  copyData(data, initialData);
```

Comment:
Also consider the data passed to hooks, e.g. "before save" hook. I think the data should not include primary key when the user is not including it.

It makes me wonder, perhaps we are hard-setting the primary key to prevent people from _changing_ primary key via `updateAttributes`? I am sure there are other ways how to implement that, without side effects like the one you are trying to fix...